src/source/geojson_wrapper.js

```diff
@@ -17,8 +17,8 @@ class FeatureWrapper {
             this.rawGeometry = feature.geometry;
         }
         this.properties = feature.tags;
-        if ('id' in feature) {
-            this.id = feature.id;
+        if ('id' in feature && !isNaN(feature.id)) {
```

Comment:
Can you add a comment here explaining the original issue and the reason for `!isNaN`/`parseInt`?